src/core/filters/meta/filter.rs

```diff
@@ -0,0 +1,467 @@
+//! # FilterMeta
+//!
+//! Object for packet filtering it implements from_string(). It takes
+//! as input a filter string under the form
+//! struct_name>member1>member2>..>leafmember generating a sequence of
```

Comment:
because '.' and "->" have a precise meaning in C notation, right?
I preferred '>' because no meaning should be given (it simply means "next nested").
AFAICT, '.' is a bit ambiguous.

Reply:
> > > > because '.' and "->" have a precise meaning in C notation, right? I preferred '>' because no meaning should be given (it simply means "next nested"). AFAICT, '.' is a bit ambiguous.
> > > 
> > > 
> > > On the other hand we're not strictly writing C and `>` also has a meaning while `.` already provides the "member of" meaning in most languages.
> > 
> > 
> > It's not just a "member of" (in terms of "offset of"), but there is indirection involved, and potentially more pointers loads/deref.
> 
> Right, but that is what happens under the hood. It's hidden from the user. Both `>` and `.` have the same "issue" (I would say "nice ux feature" though :)); so it's really a matter of if mapping to a C expectation is an issue.
> 
> > The ambiguous part is that `struct_type.something.smth_else` feels `smth_else` is at an offset of the `sk_buff` and this is not always true.
> 
> Which seems to be you point. Looking at other tools mapping a kernel object to another representation, `bpftrace` uses both `->` and `.` depending on how to access a field and `drng` uses `.`. The former does so because the code is actually written in pseudo-C and they do not provide the abstraction, while the later converts everything to `.` because it's a Python interpreter.
> 

nice idea to look at them.

> My impression is we're providing an abstraction because we don't distinguish between `->` and `.`. With that I think it's fine to select whichever symbol as it's by definition not linked to another language. But that's very questionable I agree :)
> 

Yep.
@amorenoz's preference can shed some light on this :)

> > I'm not sure, many know C (it's a kernel data structure) and honestly, nobody can have a feeling that `sk_buff>dev>name` is sk_buff is greater than dev and so forth.
> 
> Fair enough :)